### ISSUE TITLE
Replace lodash.omit for performance

### ISSUE THREAD

**Issue by chrisbolin** [FormidableLabs/victory#956]
### On What Evidence

* Lodash is dropping `omit` in the next major release [because of performance reasons](https://github.com/lodash/lodash/issues/2930). They also call out `omit`'s poor performance in [the docs](https://lodash.com/docs/4.17.5#omit): "This method is considerably slower than _.pick".

* When doing performance profiling of a Victory parallel axis chart @tptee saw that `omit` took up a considerable amount of time in the hot paths.

* Semantic-UI-React [dropped `omit`](https://github.com/Semantic-Org/Semantic-UI-React/issues/864) and **"improved the render times of all components by ~12,000 times."** (Yeah. I know.)

**Comment by chrisbolin** [comment]
### Plan

- find usage in `victory-core`, `victory-pie`, `victory-chart`, and `victory-native`

- design and profile a replacement (ideally has the same API)

- replace the thing!

**Comment by chrisbolin** [comment]
## Performance Tests

Created a [https://jsperf.com/lodash-omit-2](JS Perf test) comparing 4 approaches:

### Test Cases

#### lodash.omit

```javascript
const userWithoutPhone = _.omit(user, ["phone"]);
```

#### Object.assign and delete

```javascript
const userWithoutPhone = Object.assign({}, user);
delete userWithoutPhone["phone"];
```

#### Object literal spread (ECMAScript 2018)

```javascript
const {phone, ...userWithoutPhone} = user;
```

#### Babel's object literal spread

When Babel transpiles the below code it [creates a helper function](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=MYewdgzgLgBA3gBwBbgKYBoYDocFcKoBOA6gJZQq5QAKKYqAvjALwz5EDcQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=true&presets=es2015%2Ces2016%2Ces2017%2Creact%2Cstage-0%2Cstage-1%2Cstage-2%2Cstage-3%2Ces2015-loose&prettier=false&targets=&version=6.26.0&envVersion=) `_objectWithoutProperties`.

```javascript
// this, but run through Babel
const {phone, ...userWithoutPhone} = user;
```

### Results

No matter the browser, **Babel spread was always the fastest** and **`_.omit` was always the slowest.**

**Chrome**: Babel _10x_ faster than _.omit
**Safari**: _4.0x_ faster
**Firefox**: _2.7x_ faster

**Brower Details**

Chrome 65.0.3325
Safari 9.1.2
Firefox 58.0.0 

**OS**

Mac OS X 10.11.0

**Comment by chrisbolin** [comment]
Babel's approach is pretty optimized. I tried a variation that would only consider the Object and not the prototype:

```javascript
function objectWithoutKeys(obj, keys) {
  return Object.keys(obj).reduce((newObject, key) => {
    if (keys.indexOf(key) === -1) newObject[key] = obj[key];
    return newObject;
  }, {});
}
```

This was basically indistinguishable from Babel's `_objectWithoutProperties` (one run it was 1% faster and one run it was 1% slower). Not worth pursuing.

**Comment by chrisbolin** [comment]
resolved by FormidableLabs/victory-core#346 FormidableLabs/victory-chart#577. released April 4, 2018.
